src/asm.rs

```diff
@@ -1,254 +1,558 @@
+            // Let's just emulate it.
+            let block = self.llbb();
+            let extended_asm = block.add_extended_asm(None, "");
+            for input in inputs {
+                extended_asm.add_input_operand(None, "r", input);
+            }
+            extended_asm.add_clobber("memory");
+            extended_asm.set_volatile_flag(true);
+        } 
+        else {
+            // TODO(@Commeownist): switch to `struct_span_err_with_code` 
+            // once we get this merged into rustc
+            self.sess().struct_span_err(span, "GCC backend does not support `llvm_asm!`")
+                .help("consider using the `asm!` macro instead")
+                .emit();
+        }
+
+        // We return `true` even if we've failed to generate the asm
+        // because we want to suppress the "malformed inline assembly" error
+        // generated by the frontend.
+        true
     }
 
-    fn codegen_inline_asm(&mut self, template: &[InlineAsmTemplatePiece], operands: &[InlineAsmOperandRef<'tcx, Self>], options: InlineAsmOptions, _span: &[Span]) {
+    fn codegen_inline_asm(&mut self, template: &[InlineAsmTemplatePiece], rust_operands: &[InlineAsmOperandRef<'tcx, Self>], options: InlineAsmOptions, _span: &[Span]) {
         let asm_arch = self.tcx.sess.asm_arch.unwrap();
+        let is_x86 = matches!(asm_arch, InlineAsmArch::X86 | InlineAsmArch::X86_64);
+        let att_dialect = is_x86 && options.contains(InlineAsmOptions::ATT_SYNTAX);
+        let intel_dialect = is_x86 && !options.contains(InlineAsmOptions::ATT_SYNTAX);
 
-        let intel_dialect =
-            match asm_arch {
-                InlineAsmArch::X86 | InlineAsmArch::X86_64 if !options.contains(InlineAsmOptions::ATT_SYNTAX) => true,
-                _ => false,
-            };
+        // GCC index of an output operand equals its position in the array 
+        let mut outputs = vec![];
+
+        // GCC index of an input operand equals its position in the array
+        // added to `outputs.len()`
+        let mut inputs = vec![];
 
-        // Collect the types of output operands
-        // FIXME(antoyo): we do this here instead of later because of a bug in libgccjit where creating the
-        // variable after the extended asm expression causes a segfault:
-        // https://gcc.gnu.org/bugzilla/show_bug.cgi?id=100380
-        let mut output_vars = FxHashMap::default();
-        let mut operand_numbers = FxHashMap::default();
-        let mut current_number = 0;
-        for (idx, op) in operands.iter().enumerate() {
+        // Clobbers collected from `out("explicit register") _` and `inout("expl_reg") var => _`
+        let mut clobbers = vec![];
+
+        // We're trying to preallocate space for the template
+        let mut constants_len = 0;
+
+        // There are rules we must adhere to if we want GCC to do the right thing:
+        // 
+        // * Every local variable the asm block uses as output must be declared *before*
```

Comment:
```suggestion
        // * Every local variable the asm block used as output must be declared *before*
```

Reply:
I think the grammar is correct :)

Reply:
I kinda tend to omit extra "that"s and some articles. Lot's of my friends, including native speakers (British) do the same, so I guess it's just one of my peculiarities I have developed. Or a local thing. Or something.

Rephrased.